frappe/public/js/frappe/list/listview.js

```diff
@@ -203,21 +202,41 @@ frappe.views.ListView = Class.extend({
 			this.id_list.push(data.name);
 		}
 
-
-		var main = frappe.render_template("list_item_main", {
-			data: data,
-			columns: this.columns,
-			subject: this.get_avatar_and_id(data, true),
-			list: this,
-			right_column: this.settings.right_column
-		});
-
-		$(frappe.render_template("list_item_row", {
-			data: data,
-			main: main,
-			list: this,
-			right_column: this.settings.right_column
-		})).appendTo(row);
+		if(this.meta.image_view == 0){
+			var main = frappe.render_template("list_item_main", {
+				data: data,
+				columns: this.columns,
+				subject: this.get_avatar_and_id(data, true),
+				list: this,
+				right_column: this.settings.right_column
+			});
+	
+			$(frappe.render_template("list_item_row", {
+				data: data,
+				main: main,
+				list: this,
+				right_column: this.settings.right_column
+			})).appendTo(row);
+		}
+		else{
```

Comment:
minor: should be `else {` not `else{`

Also put this new new functions like `render_list_row` and `render_list_image`